whereby/android-sdk-demo
# Remiquinto/remove whereby room activity and broadcast receiver

Since we removed the WherebyRoomActivity component and the broadcast receiver for async events, the demo app is quite simplified: it's one single activity (no need for a navigation bar anymore). Currently, the top part of the screen provides two buttons to present the fragment either as embedded (a portion of the screen) or fullscreen; we can decide to remove the fullscreen mode if it does not add value to the demo. 

**Test plan**
1. Since the Whereby SDK version was updated: open the build.gradle file and sync the project (if needed)
2. Set the `mRoomUrlString` property (you can use `https://team.whereby.com/test-android`). 
3. Run the app on the simulator and press the `start embedded` button:
3.a. Confirm that the webview is correctly loaded
3.b. Join the room and confirm the microphone and camera button works as expected
3.c. Press the remove fragment button
4. Press the `start fullscreen` button and confirm the fragment is loaded in fullscreen
4.a. While in fullscreen, confirm that pressing the back button in the bottom left corner would remove the fullscreen fragment and show the main activity. 

As always, any comments or suggestions are welcome. 

**Gotchas**
Currently, rotating the device will reload the webview. This will be addressed separately. 

**Screenshots**
<img width="290" alt="Screenshot 2022-07-05 at 16 15 44" src="https://user-images.githubusercontent.com/7428931/177349487-183e5c09-3352-4e9a-8730-23d5a12b82a1.png">
<img width="297" alt="Screenshot 2022-07-05 at 16 16 07" src="https://user-images.githubusercontent.com/7428931/177349514-689eb299-955a-4134-8959-f27b65842af0.png">
<img width="493" alt="Screenshot 2022-07-05 at 16 17 14" src="https://user-images.githubusercontent.com/7428931/177349526-80dd3e86-261f-4226-804e-d9a8fd461f81.png">
<img width="495" alt="Screenshot 2022-07-05 at 16 17 00" src="https://user-images.githubusercontent.com/7428931/177349542-d7c64f26-cd7a-4740-a9f4-277b1e711ca2.png">
